lib/rules/onClick.js

```diff
@@ -41,19 +41,30 @@ module.exports = {
                 if (bypass) return;
 
                 context.report({
-                  node,
-                  message: getError(errors.onClick.message, testAttribute),
-                  fix: function fix(fixer) {
-                    const { nanoid } = require('nanoid');
-                    const suggestedId = nanoid();
-                    const start = node.range[0] + node.name.name.length;
-                    const end = start + 1;
-
-                    return fixer.insertTextAfterRange(
-                      [start, end],
-                      ' ' + testAttribute + '="' + suggestedId + '"'
-                    );
-                  },
+                    node,
+                    message: getError(errors.onClick.message, testAttribute),
+                    fix: function fix(fixer) {
+                        const { nanoid } = require('nanoid');
+                        const suggestedId = nanoid();
+                        const namePositionEnd = node.name.range[1];
+                        const isSingleLine = node.loc.start.line === node.loc.end.line;
```

Comment:
interesting, some single line detection- cool. can we get a unit test for this, as well as one exercising the "." in a node?

you can add some entries into this test array: https://github.com/cision/eslint-plugin-test-selectors/blob/master/tests/lib/rules/onClick.js#L78

Reply:
Yeah... Thanks for being a good conscience.

Reply:
Done!